share/wake/lib/system/job.wake

```diff
@@ -202,6 +202,129 @@ export def localRunner: Runner =
     if plan.getPlanLocalOnly then Pass 1.0 else Fail "cannot detect outputs"
   Runner "local" score doit
 
+# TODO: Delete these once new json API is added to wake repo
+def jField (jvalue: JValue) (key: String) =
+  require JObject obj = jvalue
+  else failWithError "not an object"
+
+  require Some (Pair (Pair _ value) _) = find (_.getPairFirst ==~ key) obj
+  else failWithError "key '{key}' not found"
+
+  Pass value
+
+def jInteger = match _
+  Pass (JInteger x) = Pass x
+  Fail err          = Fail err
+  _                 = failWithError "not an integer"
+
+def jString = match _
+  Pass (JString x) = Pass x
+  Fail err         = Fail err
+  _                = failWithError "not a string"
+
+def jDouble = match _
+  Pass (JDouble x) = Pass x
+  Fail err         = Fail err
+  _                = failWithError "not a number"
+
+def jArray (fn: Result JValue Error => Result a Error) = match _
+  Pass (JArray arr) = map (fn _.Pass) arr | findFail
+  Fail err          = Fail err
+  _                 = failWithError "not an array"
+
+export def mkJobCacheRunner ((Runner name score baseDoIt): Runner): Runner =
+  def virtual job stdout stderr status runtime cputime membytes ibytes obytes = prim "job_virtual"
+  def badlaunch job error = prim "job_fail_launch"
+  def job_cache_read str = prim "job_cache_read"
+  def job_cache_add str = prim "job_cache_add"
+
+  def doit job runnerInput = match runnerInput
+    Fail e =
+      def _ = badlaunch job e
+      Fail e
+    Pass (RunnerInput _label cmd vis env dir stdin _ _ _) =
+        def mkVisJson (Path path hash) =
+          JObject ("path" :-> JString path, "hash" :-> JString hash,)
+        def jobCacheVisible = JArray (map mkVisJson vis)
+        def jobCacheJsonIn = prettyJSON $ JObject (
+          "cwd" :-> JString dir,
+          "command_line" :-> JString cmd.implode,
+          "envrionment" :-> JString env.implode,
```

Comment:
`envrionment` is spelt this way a few times

Reply:
💀 

I'll fix that soon